libs/tup-components/src/tickets/TicketModal.tsx

```diff
@@ -0,0 +1,379 @@
+                <tr {...row.getRowProps()}>
+                  {row.cells.map((cell) => (
+                    <td {...cell.getCellProps()}>{cell.render('Cell')}</td>
+                  ))}
+                </tr>
+              );
+            })
+          ) : (
+            <tr>
+              <td colSpan={5}>No attachments to display.</td>
+            </tr>
+          )}
+        </tbody>
+      </table>
+    </div>
+  );
+};
+Attachments.propTypes = {
+  attachments: PropTypes.arrayOf(PropTypes.array.isRequired).isRequired,
+  ticketId: PropTypes.string.isRequired,
+};
+
+const TicketHistoryReply: React.FC<{ ticketId: string }> = ({ ticketId }) => {
+  const mutation = useTicketReply(ticketId);
+  const { mutate, isLoading, isError } = mutation;
+
+  const defaultValues = useMemo(
+    () => ({
+      text: '',
+      files: [],
+    }),
+    []
+  );
+
+  return (
+    <Formik
+      enableReinitialize
+      initialValues={defaultValues}
+      validationSchema={formSchema}
+      onSubmit={(values, { resetForm }) => {
+        const formData = new FormData();
+        formData.append('text', values['text']);
+        if (values.files) {
+          values.files.forEach((file) => formData.append('files', file));
+        }
+        mutate(formData, {
+          onSuccess: () => resetForm(),
+        });
+      }}
+    >
+      {({ isSubmitting, isValid }) => {
+        return (
+          <Form className="ticket-reply-form">
+            <FormikInput
+              name="text"
+              label="Reply"
+              type="textarea"
+              className="ticket-reply-text-area"
+              description=""
+              required
+            />
+            <FileInputDropZoneFormField
+              id="files"
+              isSubmitted={isSubmitting}
+              description="Error reports and screenshots can be helpful for diagnostics"
+              maxSizeMessage="Max File Size: 3MB"
+              maxSize={3145728}
+            />
+            <FormGroup className="ticket-reply-submission">
+              {isError && (
+                <InlineMessage type="error">
+                  Something went wrong.
+                </InlineMessage>
+              )}
+              <Button
+                attr="submit"
+                type="primary"
+                disabled={!isValid || isSubmitting || isLoading || isError}
+                isLoading={isLoading}
+              >
+                Reply
+              </Button>
+            </FormGroup>
+          </Form>
+        );
+      }}
+    </Formik>
+  );
+};
+
+TicketHistoryReply.propTypes = {
+  ticketId: PropTypes.string.isRequired,
+};
+
+const TicketHistoryCard: React.FC<historyCardParams> = ({
+  created,
+  creator,
+  isCreator,
+  content,
+  attachments,
+  ticketId,
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const toggleIcon = isOpen ? (
+    <i className="icon-action icon-collapse" />
+  ) : (
+    <i className="icon-action icon-expand" />
```

Comment:
**Request.** Please use @tacc/core-components `<Icon>` (example is on line [#273](https://github.com/TACC/tup-ui/pull/76/files#diff-9674095a67d00cd54623636b0c0105677703fbb318b3fe0f89944207603d2349R273)).

Reply:
`icon-expand` is showing up, but I think we may be missing **`icon-collapse`**.

Reply:
Thanks. I'll ask designers for this.

A placeholder fix (untested global code):
```
.icon-collapse {
  content: "\ea24"; /* copied from .icon-expand::before */
  transform: rotate(180deg);
}
```

<details><summary>Source</summary>

- quickie fix: https://github.com/TACC/Core-Styles/blob/v0.11.0/src/lib/_imports/components/cortal.icon.css#L68-L70
- careful alignment: https://github.com/TACC/Core-Styles/blob/v1.0.0/src/lib/_imports/components/cortal.icon.css#L68

</details>

Reply:
https://github.com/TACC/tup-ui/pull/76/files#r1046266648